framework/OutStreams/OutStreamInterface.py

```diff
@@ -0,0 +1,117 @@
+    spec.addSub(InputData.parameterInputFactory('filename', contentType=InputTypes.StringType,
+        descr=r"""Sets the name to use for the saved file."""))
+    return spec
+
+  def __init__(self):
+    """
+      Init of Base class
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'OutStreamInterface'
+    self.overwrite = True       # overwrite existing creations?
+    self.subDirectory = None    # directory to save generated files to
+    self.filename = ''          # target file name
+    self.numberAggregatedOS = 1 # number of aggregated outstreams # no addl info from original OutStream
+
+  def handleInput(self, spec):
+    """
+      Loads the input specs for this object.
+      @ In, spec, InputData.ParameterInput, input specifications
+      @ Out, None
+    """
+    super().handleInput(spec)
+    self.subDirectory = spec.parameterValues.get('dir')
+    self.overwrite = spec.parameterValues.get('overwrite')
+    fname = spec.findFirst('filename')
+    if fname is not None:
+      self.filename = fname
+
+  def initialize(self, stepEntities):
+    """
+      Initialize for a new Step
+      @ In, stepEntities, dict, Entities from the Step
+      @ Out, None
+    """
+    super().initialize()
+    if self.subDirectory is not None:
+      if not os.path.exists(self.subDirectory):
+        os.makedirs(self.subDirectory)
+
+  # this method is used by the original OutStreams, GeneralPlot and File Print.
+  def legacyCollectSources(self, inDict):
+    """
+      Collect the usable sources in the format this plotter expects.
+      @ In, inDict, dict, Step entities
+      @ Out, None
+    """
+    self.sourceData = []
+    for outIndex in range(self.numberAggregatedOS):
+      foundData = False
+      for output in inDict['Output']:
+        if output.name.strip() == self.sourceName[outIndex] and output.type in DataObjects.factory.knownTypes():
+          self.sourceData.append(output)
+          foundData = True
+          break
+      if not foundData:
+        for inp in inDict['Input']:
+          if not type(inp) == type(""):
+            if inp.name.strip() == self.sourceName[outIndex] and inp.type in DataObjects.factory.knownTypes():
+              self.sourceData.append(inp)
+              foundData = True
+            elif type(inp) == Models.ROM:
+              self.sourceData.append(inp)
+              foundData = True  # good enough
+      if not foundData and 'TargetEvaluation' in inDict.keys():
+        if inDict['TargetEvaluation'].name.strip() == self.sourceName[outIndex] and inDict['TargetEvaluation'].type in DataObjects.factory.knownTypes():
+          self.sourceData.append(inDict['TargetEvaluation'])
+          foundData = True
+      if not foundData and 'SolutionExport' in inDict.keys():
+        if inDict['SolutionExport'].name.strip() == self.sourceName[outIndex] and inDict['SolutionExport'].type in DataObjects.factory.knownTypes():
```

Comment:
I personally would newline these conditionals. To me, they get harder to read the longer they are.

Reply:
Inherited, but worth changing. Thanks.